amf-shapes/shared/src/main/scala/amf/shapes/client/platform/model/domain/jsonldinstance/JsonLDObject.scala

```diff
@@ -4,13 +4,38 @@ import amf.core.client.platform.model.domain.DomainElement
 import amf.shapes.client.scala.model.domain.jsonldinstance.{JsonLDObject => InternalJsonLDObject}
 import amf.shapes.internal.domain.metamodel.jsonldschema.JsonLDEntityModel
 import amf.shapes.internal.spec.jsonldschema.parser.JsonPath
+import amf.shapes.internal.convert.ShapeClientConverters._
 
 import scala.scalajs.js.annotation.{JSExportAll, JSExportTopLevel}
 
 @JSExportAll
 class JsonLDObject(override private[amf] val _internal: InternalJsonLDObject) extends DomainElement with JsonLDElement {
   @JSExportTopLevel("JsonLDObject")
-  def this() = this(InternalJsonLDObject.empty(new JsonLDEntityModel(Nil, Nil, JsonPath.empty), JsonPath.empty))
+  def this() = this(InternalJsonLDObject.empty(JsonLDEntityModel(Nil, Nil, JsonPath.empty), JsonPath.empty))
   def componentId: String = _internal.componentId
 
+  def withProperty(property: String, value: String): JsonLDObject = _internal.withProperty(property, value)
+
+  def withProperty(property: String, value: Integer): JsonLDObject = _internal.withProperty(property, value)
+
+  def withProperty(property: String, value: Float): JsonLDObject = _internal.withProperty(property, value)
+
+  def withProperty(property: String, value: Boolean): JsonLDObject = _internal.withProperty(property, value)
+
+  def withProperty(property: String, value: JsonLDObject): JsonLDObject = _internal.withProperty(property, value)
+
+  def withStringPropertyCollection(property: String, values: Seq[String]): JsonLDObject =
```

Comment:
can't we use method overloading instead of having a different name ?

Reply:
That was my original approach, but compilation failed :S